### PR TITLE
Feature/accessibility support

### PR DESCRIPTION
**Pull request by simonmitchell**
Adds accessibility refresh to table view controller, and support for darker text colours to `Theme`

### CODE REVIEW

File: ThunderTable/TableViewController.swift [hunk 1]
```diff
@@ -136,26 +136,71 @@ open class TableViewController: UITableViewController {
...
     }
 	
-	private var dynamicChangeObserver: NSObjectProtocol?
-	
-	public var shouldRedrawWithContentSizeChange = true
+    private var dynamicChangeObserver: NSObjectProtocol?
+    
+    private var accessibilityObservers: [Any] = []
+    
+    public var adjustsFontForContentSizeCategory: Bool = true
```
Comment (ryanbourneuk) [inline ThunderTable/TableViewController.swift:144]: Docs for these please :D

File: ThunderTable/TableViewController.swift [hunk 2]
```diff
@@ -136,26 +136,71 @@ open class TableViewController: UITableViewController {
...
-			guard let strongSelf = self, strongSelf.shouldRedrawWithContentSizeChange else { return }
-			strongSelf.tableView.reloadData()
+		dynamicChangeObserver = NotificationCenter.default.addObserver(forName: UIContentSizeCategory.didChangeNotification, object: nil, queue: .main) { [weak self] (notification) in
+			guard let strongSelf = self, strongSelf.adjustsFontForContentSizeCategory else { return }
+            strongSelf.reloadVisibleRowsWhilstMaintainingSelection()
+            strongSelf.accessibilitySettingsDidChange()
 		}
+        
+        // Notification names that it makes sense to redraw on
+        let accessibilityNotifications: [Notification.Name] = [
```
Comment (ryanbourneuk) [inline ThunderTable/TableViewController.swift:169]: Is this a duplication of `accessibilityRedrawNotificationNames` (just with a few more) - what's the reasoning behind this? :)

File: ThunderTable/TableViewController.swift (hunk 1 above)
Comment (simonmitchell) [reply ThunderTable/TableViewController.swift:144]: Docs should be available from the protocol that this is a variable from I believe :D

Comment (simonmitchell) [reply ThunderTable/TableViewController.swift:144]: Checked, and apparently that is not true!

File: ThunderTable/TableViewController.swift (hunk 2 above)
Comment (simonmitchell) [reply ThunderTable/TableViewController.swift:169]: See other (ThunderCloud) PR. This is so we don't have to have messy code and risk double-adding (or removing) these observers in the `didSet` for `accessibilityRedrawNotificationNames`. We'd also have to keep track of whether the VC is visible and use that in the `didSet` method. I figured it was easier to just listen to all notifications and then check in the callback as to whether the given notification name is present in `strongSelf.accessibilityRedrawNotificationNames`

**Review by ryanbourneuk — APPROVED** [review]

**Pull request merged**